### PR TITLE
🗾 Link map and list view and consolidate querying logic

### LINKED ISSUE #78: Consolidate querying logic in Filter and MapPage based on SearchContext

### LINKED ISSUE #83: Fix primary text color in ListEntry
Search results primary text color is not right

### LINKED ISSUE #117: Don’t reload list view when map view is active on mobile
## Learning objectives

- Objective

## Todo

These points are a rough guideline. **Please** feel free to discuss with others on the team about the best way to design this component!

- Todo

## Useful links

- Useful link

### PR DESCRIPTION
**Pull request by Qrtn**
## Status:

:rocket: Ready

## Description

- Links map and list view
- Moves ownership of current tab index to router/pathname
- Get rid of overview and reviews having two loading indicators...
- Move PhotoGrid above tabs
- **Consolidate querying logic in getFilteredParams**

Fixes #78, fixes #117, fixes #83

## Todos

- Further state management improvement will occur with move to Redux

## Screenshots

image

### CODE REVIEW

**Comment by Qrtn** [comment]
Feel free to review after first PR is merged in.

File: src/components/entry/Entry.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    if (typesById) {
+      fetchEntryData()
+    }
+  }, [id, typesById])
+
+  let content
+  if (!locationData || !reviews) {
+    content = <LoadingIndicator vertical cover />
+  } else {
+    const entryOverview = <EntryOverview locationData={locationData} />
```
Comment (SirajChokshi) [inline src/components/entry/Entry.js:70]: Are these constants actually useful?

**Review by SirajChokshi — APPROVED** [review]
Good work!

File: src/components/entry/Entry.js (hunk 1 above)
Comment (Qrtn) [reply src/components/entry/Entry.js:70]: It's mainly so that they can be used again in 94-96.

**Pull request merged**